spring-boot-admin-client/src/test/java/de/codecentric/boot/admin/client/registration/RegistrationApplicationListenerTest.java

```diff
@@ -38,22 +40,27 @@
     @Test
     public void test_register() {
         ApplicationRegistrator registrator = mock(ApplicationRegistrator.class);
-        TaskScheduler scheduler = mock(TaskScheduler.class);
-        RegistrationApplicationListener listener = new RegistrationApplicationListener(registrator, scheduler);
+        RegistrationApplicationListener listener = new RegistrationApplicationListener(registrator);
+
+        TaskScheduler scheduler = mock(ThreadPoolTaskScheduler.class);
+        ReflectionTestUtils.setField(listener, "taskScheduler", scheduler);
```

Comment:
I think it would be more suitable to have a package-private constructor to inject the mock so we don't rely on reflection.